Watchman.Engine/Generation/Sqs/SqsResourceAlarmGenerator.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Amazon.DynamoDBv2.Model;
+using Watchman.AwsResources;
+using Watchman.AwsResources.Services.Sqs;
+using Watchman.Configuration.Generic;
+using Watchman.Engine.Alarms;
+
+namespace Watchman.Engine.Generation.Sqs
+{
+    public class SqsResourceAlarmGenerator : IResourceAlarmGenerator<QueueData, SqsResourceConfig>
+    {
+        private readonly AlarmBuilder<QueueData, SqsResourceConfig> _builder;
+        private readonly IResourceSource<QueueData> _queueSource;
+        private readonly IAlarmDimensionProvider<QueueData> _dimensionProvider;
+        private readonly IList<AlarmDefinition> _errorQueueDefaults = Defaults.SqsError;
+
+        public SqsResourceAlarmGenerator(
+            IResourceSource<QueueData> queueSource,
+            IAlarmDimensionProvider<QueueData> dimensionProvider,
+            IResourceAttributesProvider<QueueData, SqsResourceConfig> attributeProvider)
+        {
```

Comment:
I probably have misunderstood, but does the implementation mean that the regex needs to match both the queue and the error queue. And if I specify queue by name (not regex) how would it pick up the error queue?

Reply:
for example if I specify queue as
pattern: ^queue-name$

or name: queue-name

would neither of them match queue-name_Error and therefore it wouldn't be in the expanded set of resources?

Reply:
Hmmm I think you're right the current implementation is broken for named queues. Will require some more thought I think.

Reply:
I wonder if a better solution might be to have the `QueueSource` return a different DTO that contains the primary queue and its error queue. 

That way the builder could work in a similar way as for dynamodb where it creates alarm(s) against the primary resource (table) and secondary resource (gsi) but in our case it's a bit simpler because both resources are queue.

so instead of `QueueData[]` the queue source might return a list of something that contains 
```
e.g. QueueAndErrorQueue
{
   QueueData Queue
   QueueData ErrorQueue
}
```

the queue source would then only return error queues in the error property and normal queues in the queue property

Reply:
hang on, re-read, this is what we already have in master. QueueData has an ErrorQueue property. So can't we just use that in the builder?